pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -573,6 +573,13 @@ func setH2Options(cluster *cluster.Cluster) {
 
 func applyTrafficPolicy(opts buildClusterOpts) {
 	connectionPool, outlierDetection, loadBalancer, tls := selectTrafficPolicyComponents(opts.policy)
+	if opts.direction == model.TrafficDirectionOutbound && connectionPool != nil && connectionPool.Http != nil && connectionPool.Http.UseClientProtocol {
+		setH2Options(opts.cluster)
```

Comment:
not sure why it is here? IIUC `applyH2Upgrade` has already done this

Reply:
applyH2Upgrade is not applicable when useClientProtocol is set.